src/fiskaltrust.Middleware.Demo/RestPos.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            throw new NotImplementedException();
+        }
+
+        IAsyncResult ifPOS.v0.IPOS.BeginSign(ifPOS.v0.ReceiptRequest data, AsyncCallback callback, object state)
+        {
+            throw new NotImplementedException();
+        }
+
+        string ifPOS.v0.IPOS.Echo(string message)
+        {
+            throw new NotImplementedException();
+        }
+
+        async Task<EchoResponse> ifPOS.v1.IPOS.EchoAsync(EchoRequest message)
+        {
+            if (_url.StartsWith("rest://"))
+            {
+                return await JsonEchoAsync(message);
+            }
+            else
+            {
+                return await XmlEchoAsync(message);
+            }
+
+        }
+
+        private async Task<EchoResponse> XmlEchoAsync(EchoRequest message)
+        {
+            var jsonstring = JsonConvert.SerializeObject(message);
+            var xmlContent = new StringContent(jsonstring, Encoding.UTF8, "application/xml");
+
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(_url);
+
+                using (var response = await client.PostAsync("v1/Echo", xmlContent))
+                {
+                    var content = await response.Content.ReadAsStringAsync();
+
+                    var xml = XElement.Parse(content);
+                    string jsonText = JsonConvert.SerializeXNode(xml);
+                    return JsonConvert.DeserializeObject<EchoResponse>(jsonText);
+                }
+            }
+        }
+
+        private async Task<EchoResponse> JsonEchoAsync(EchoRequest message)
+        {
+            var jsonstring = JsonConvert.SerializeObject(message);
+            var jsonContent = new StringContent(jsonstring, Encoding.UTF8, "application/json");
+
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(_url);
+
+                using (var response = await client.PostAsync("v1/Echo", jsonContent))
+                {
+                    var content = await response.Content.ReadAsStringAsync();
+                    return JsonConvert.DeserializeObject<EchoResponse>(content.ToString());
+                }
+            }
+        }
+
+        string ifPOS.v0.IPOS.EndEcho(IAsyncResult result)
+        {
+            throw new NotImplementedException();
+        }
+
+        Stream ifPOS.v0.IPOS.EndJournal(IAsyncResult result)
+        {
+            throw new NotImplementedException();
+        }
+
+        ifPOS.v0.ReceiptResponse ifPOS.v0.IPOS.EndSign(IAsyncResult result)
+        {
+            throw new NotImplementedException();
+        }
+
+        Stream ifPOS.v0.IPOS.Journal(long ftJournalType, long from, long to)
+        {
+            throw new NotImplementedException();
+        }
+
+        IAsyncEnumerable<JournalResponse> ifPOS.v1.IPOS.JournalAsync(JournalRequest request)
+        {
+            throw new NotImplementedException();
```

Comment:
I would change this to a `NotSupportedException` and add the details as error message (Because JournalAsync works with streams and asyncstreams are not supported in http)